phangsPipeline/handlerImaging.py

```diff
@@ -829,175 +1005,223 @@ def recipe_phangsalma_imaging(
-            target = target, config = config, product = product, 
-            extra_ext_in = extra_ext_in, suffix_in = suffix_in,
-            extra_ext_out = extra_ext_out,
-            stage = 'multiscale')
+            target=target, config=config, product=product,
+            imaging_method=imaging_method,
+            extra_ext_in=extra_ext_in, suffix_in=suffix_in,
+            extra_ext_out=extra_ext_out,
+            stage='multiscale')
 
         # AKL - adding a check on the mask existence here
 
         # Check if we have a mask in place. This might happen if we
         # didn't find a clean mask or if we did not supply one. Either
         # way, now turn on a primary beam based mask.
 
-        fname_dict = self._fname_dict(product = product, imagename = clean_call.get_param('imagename'))
-        if os.path.isdir(fname_dict['mask']) == False:
-            logger.warning("I looked for this and did not find it: "+fname_dict['mask'])
-            logger.warning("No mask found, we will use a pblimit mask.")
-            clean_call.set_param('usemask', 'pb')
-            if clean_call.get_param('pblimit') is None:
-                clean_call.set_param('pbmask', 0.5)
+        fname_dict = self._fname_dict(product=product, imagename=clean_call.get_param('imagename'),
+                                      imaging_method=imaging_method)
+
+        mask_name = fname_dict['mask']
+        if not os.path.isdir(mask_name):
+            logger.warning("I looked for this and did not find it: " + mask_name)
+            if (clean_call.get_param('usemask') == "user") and (clean_call.get_param('mask') is not None) and (
+                    clean_call.get_param('mask') != ''):
+                logger.debug("Using the user mask \"%s\" as defined in the clean parameter file." % (
+                    clean_call.get_param('mask')))
             else:
-                clean_call.set_param('pbmask', clean_call.get_param('pblimit'))
+                logger.warning("No mask found, we will use a pblimit mask.")
+                clean_call.set_param('usemask', 'pb')
+                if clean_call.get_param('pblimit') is None:
+                    clean_call.set_param('pbmask', 0.5)
+                else:
+                    clean_call.set_param('pbmask', clean_call.get_param('pblimit'))
 
         # AKL - propose to deprecate
-        #if not do_read_clean_mask: 
+        # if not do_read_clean_mask:
         #    clean_call.set_param('usemask', 'pb')
         #    clean_call.set_param('pbmask', 0.2)
-        
+
         # Dynamic sizing
-        
+
         if dynamic_sizing:
-            clean_call.set_param('cell',cell,nowarning=True)
-            clean_call.set_param('imsize',imsize,nowarning=True)                    
+            clean_call.set_param('cell', cell, nowarning=True)
+            clean_call.set_param('imsize', imsize, nowarning=True)
+
+        # Cube params for sdintimaging
+
+        if imaging_method == 'sdintimaging':
```

Comment:
They need to be changed because they need to match the SD cube *exactly*. For some reason trying to pull this out manually sometimes misses the direction of the frequency increments, so there's a convenience task from sdintimaging to pull these out for you. They do need to be set in the clean call though, so that's done here